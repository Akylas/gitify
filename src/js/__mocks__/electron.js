window.Notification = function (title, options) {
  this.title = title;

  return {
    onclick: jest.fn()
  };
};

window.Audio = class Audio {
  constructor(path) {
    this.path = path;
  }

  play() {}
};

window.localStorage = {
  store: {},
  getItem: function (key) { return this.store[key]; },
  setItem: function (key, item) {
    this.store[key] = item;
  },
  removeItem: jest.fn()
};

window.alert = jest.fn();

const browserWindow = {
  loadURL: jest.fn(),
  webContents: {
    on: () => {},
  },
  on: () => {},
  close: jest.fn(),
  destroy: jest.fn()
};

module.exports = {
  remote: {
    BrowserWindow: () => browserWindow
  },
  ipcRenderer: {
    send: jest.fn()
  },
  shell: {
    openExternal: jest.fn()
  },
};
