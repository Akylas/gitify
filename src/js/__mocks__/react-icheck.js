import React from 'react';

export class Checkbox extends React.PureComponent {
  render() {
    return (
      <div>
        <strong>{this.props.label}</strong>
        <input
          type="checkbox"
          checked={this.props.defaultChecked}
          onChange={(evt) => this.props.onChange(evt)} />
      </div>
    );
  };
}

export class Radio extends React.PureComponent {
  render() {
    return (
      <div>
        <strong>{this.props.label}</strong>
        <input
          type="radio"
          value={this.props.value}
          checked={this.props.defaultChecked}
          onChange={(evt) => this.props.onChange(evt)} />
      </div>
    );
  };
}

export class RadioGroup extends React.PureComponent {
  render() {
    return (
      <div>
        {this.props.children}
      </div>
    );
  };
}
