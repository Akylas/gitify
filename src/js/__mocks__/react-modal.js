import React from 'react';

export default class Modal extends React.PureComponent {
  render() {
    return this.props.isOpen ? (
      <div>{this.props.children}</div>
    ) : null;
  };
}
