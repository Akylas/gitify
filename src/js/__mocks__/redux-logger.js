module.exports = function createLogger () {
  return ({ getState }) => (next) => (action) => {
    return next(action);
  };
};
