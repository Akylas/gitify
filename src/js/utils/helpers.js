const ipcRenderer = window.require('electron').ipcRenderer;

export default {
  updateTrayIcon(notificationsLength) {
    if (notificationsLength > 0) {
      ipcRenderer.send('update-icon', 'TrayActive');
    } else {
      ipcRenderer.send('update-icon', 'TrayIdle');
    }
  },

  generateGitHubUrl(url) {
    var newUrl = url.replace('api.github.com/repos', 'www.github.com');

    if (newUrl.indexOf('/pulls/') !== -1) {
      newUrl = newUrl.replace('/pulls/', '/pull/');
    }

    if (newUrl.indexOf('/releases/') !== -1) {
      newUrl = newUrl.replace('/repos', '');
      newUrl = newUrl.substr(0, newUrl.lastIndexOf('/'));
    }

    return newUrl;
  }
};
