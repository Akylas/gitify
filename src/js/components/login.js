const { ipcRenderer } = require('electron');

import React from 'react';
import { Redirect, Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { authGithub, isUserEitherLoggedIn } from '../utils/helpers';

export class LoginPage extends React.Component {
  componentWillReceiveProps(nextProps) {
    if (nextProps.isEitherLoggedIn) {
      ipcRenderer.send('reopen-window');
    }
  }

  render() {
    if (this.props.isEitherLoggedIn) {
      return (
        <Redirect to="/" />
      );
    }

    return (
      <div className="container-fluid main-container login">
        <img className="img-responsive logo" src="images/gitify-logo-outline-dark.png" />
        <div className="desc">GitHub Notifications<br />in your menu bar.</div>
        <button
          className="btn btn-block btn-login"
          onClick={() => authGithub(undefined, this.props.dispatch)}
        >
          <i className="fa fa-github mr-2" /> Login to GitHub
        </button>

        <Link
          to="/enterpriselogin"
          className="btn btn-block btn-login mt-3">
          <i className="fa fa-github mr-2" /> Login to GitHub Enterprise
        </Link>
      </div>
    );
  }
};

export function mapStateToProps(state) {
  return {
    isEitherLoggedIn: isUserEitherLoggedIn(state.auth),
  };
};

export default connect(mapStateToProps)(LoginPage);
