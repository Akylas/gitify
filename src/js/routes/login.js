const { ipcRenderer } = require('electron');
import PropTypes from 'prop-types';

import React from 'react';
import { Redirect, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGithub } from '@fortawesome/free-brands-svg-icons';

import { authGithub, isUserEitherLoggedIn } from '../utils/helpers';
import LogoDark from '../components/logos/dark';

export class LoginPage extends React.Component {
  static propTypes = {
    isEitherLoggedIn: PropTypes.bool.isRequired,
    dispatch: PropTypes.func.isRequired,
  };

  componentWillReceiveProps(nextProps) {
    if (nextProps.isEitherLoggedIn) {
      ipcRenderer.send('reopen-window');
    }
  }

  render() {
    if (this.props.isEitherLoggedIn) {
      return <Redirect to="/" />;
    }

    return (
      <div className="container-fluid main-container login">
        <LogoDark className="mt-5" />

        <div className="desc">
          GitHub Notifications
          <br />
          in your menu bar.
        </div>
        <button
          className="btn btn-block btn-login"
          onClick={() => authGithub(undefined, this.props.dispatch)}
        >
          <FontAwesomeIcon className="mr-2" icon={faGithub} title="GitHub" />{' '}
          Login to GitHub
        </button>

        <Link to="/enterpriselogin" className="btn btn-block btn-login mt-3">
          <FontAwesomeIcon className="mr-2" icon={faGithub} title="GitHub" />{' '}
          Login to GitHub Enterprise
        </Link>
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {
    isEitherLoggedIn: isUserEitherLoggedIn(state.auth),
  };
}

export default connect(mapStateToProps)(LoginPage);
