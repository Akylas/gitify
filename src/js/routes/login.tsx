const { ipcRenderer } = require('electron');

import * as React from 'react';
import { Redirect, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGithub } from '@fortawesome/free-brands-svg-icons';

import { authGithub, isUserEitherLoggedIn } from '../utils/helpers';
import { LogoDark } from '../components/logos/dark';

interface IProps {
  isEitherLoggedIn: boolean;
  dispatch: any;
}

export class LoginPage extends React.Component<IProps> {
  constructor(props: IProps) {
    super(props);

    this.state = {
      isEitherLoggedIn: props.isEitherLoggedIn,
    };
  }

  static getDerivedStateFromProps(props, state) {
    if (props.isEitherLoggedIn) {
      ipcRenderer.send('reopen-window');
    }
    return {
      isEitherLoggedIn: props.isEitherLoggedIn,
    };
  }

  render() {
    if (this.props.isEitherLoggedIn) {
      return <Redirect to="/" />;
    }

    return (
      <div className="container-fluid main-container login">
        <LogoDark className="mt-5" />

        <div className="desc">
          GitHub Notifications
          <br />
          in your menu bar.
        </div>
        <button
          className="btn btn-block btn-login"
          onClick={() => authGithub(undefined, this.props.dispatch)}
        >
          <FontAwesomeIcon className="mr-2" icon={faGithub} title="GitHub" />{' '}
          Login to GitHub
        </button>

        <Link to="/enterpriselogin" className="btn btn-block btn-login mt-3">
          <FontAwesomeIcon className="mr-2" icon={faGithub} title="GitHub" />{' '}
          Login to GitHub Enterprise
        </Link>
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {
    isEitherLoggedIn: isUserEitherLoggedIn(state.auth),
  };
}

export default connect(mapStateToProps)(LoginPage);
