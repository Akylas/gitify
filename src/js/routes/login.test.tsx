import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Map } from 'immutable';
import { MemoryRouter } from 'react-router';
import { render, fireEvent } from '@testing-library/react';

import { mockedEnterpriseAccounts } from '../__mocks__/mockedData';

const { ipcRenderer, remote } = require('electron');
const BrowserWindow = remote.BrowserWindow;

import { LoginPage, mapStateToProps } from './login';
import * as helpers from '../utils/helpers';

jest.mock('@fortawesome/react-fontawesome', () => ({
  FontAwesomeIcon: props => {
    const mockProps = {
      onClick: props.onClick,
      title: props.title,
    };

    return <div {...mockProps}>FAIcon{props.title}</div>;
  },
}));

describe('components/login.js', () => {
  const props = {
    dispatch: jest.fn(),
  };

  beforeEach(function() {
    // @ts-ignore
    new BrowserWindow().loadURL.mockReset();
    spyOn(ipcRenderer, 'send');
    props.dispatch.mockReset();
  });

  it('should test the mapStateToProps method', () => {
    const state = {
      auth: Map({
        token: '123456',
        failed: false,
        isFetching: false,
        enterpriseAccounts: mockedEnterpriseAccounts,
      }),
    };

    const mappedProps = mapStateToProps(state);

    expect(mappedProps.isEitherLoggedIn).toBeTruthy();
  });

  it('should render itself & its children', () => {
    const caseProps = {
      ...props,
      isEitherLoggedIn: false,
    };

    const tree = TestRenderer.create(
      <MemoryRouter>
        <LoginPage {...caseProps} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  it('should redirect to notifications once logged in', () => {
    const caseProps = {
      ...props,
      isEitherLoggedIn: false,
    };

    const { rerender } = render(
      <MemoryRouter>
        <LoginPage {...caseProps} />
      </MemoryRouter>
    );

    rerender(
      <MemoryRouter>
        <LoginPage {...caseProps} isEitherLoggedIn={true} />
      </MemoryRouter>
    );

    expect(ipcRenderer.send).toHaveBeenCalledTimes(1);
    expect(ipcRenderer.send).toHaveBeenCalledWith('reopen-window');
  });

  it('should call the authGitHub helper when pressing the login button', () => {
    spyOn(helpers, 'authGithub');
    const caseProps = {
      ...props,
      isEitherLoggedIn: false,
    };

    const { getByRole } = render(
      <MemoryRouter>
        <LoginPage {...caseProps} />
      </MemoryRouter>
    );

    fireEvent.click(getByRole('button'));

    expect(helpers.authGithub).toHaveBeenCalledTimes(1);
  });
});
