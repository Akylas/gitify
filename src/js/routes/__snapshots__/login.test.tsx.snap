// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/login.tsx should render itself & its children 1`] = `
<div
  className="sc-bdVaJa jZnlmR"
>
  <svg
    className="logo sc-bwzfXH kqSyDw"
    onClick={[Function]}
    role="logo"
    viewBox="0 0 500 500"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <linearGradient
        id="linearGradient-1"
        x1="90.8779989%"
        x2="-1.30249747%"
        y1="80.0676288%"
        y2="32.7108237%"
      >
        <stop
          offset="0%"
          stopColor="#22283B"
        />
        <stop
          offset="100%"
          stopColor="#555B6E"
        />
      </linearGradient>
    </defs>
    <g
      fill="none"
      fillRule="evenodd"
      id="Artboard"
      stroke="none"
      strokeWidth="1"
    >
      <g
        fill="url(#linearGradient-1)"
        fillRule="nonzero"
        id="Logo"
        transform="translate(6.000000, 6.000000)"
      >
        <path
          d="M330.391053,10.2526316 C361.413773,10.2526316 386.562656,35.4015165 386.562656,66.4242368 C386.562656,97.4469571 361.413773,122.595841 330.391053,122.595842 L330.365421,122.570211 L247.319105,122.570211 C180.042515,122.499128 125.399186,176.890222 125.159,244.166421 L125.133789,244.219632 L125.154247,245.84774 C126.219609,301.255958 164.473457,349.116728 218.453292,362.268048 C272.940509,375.542983 329.466993,350.045988 355.583747,300.428701 L272.258632,300.427737 C257.346614,300.427737 243.046846,294.496977 232.512078,283.942976 C221.97731,273.388975 216.072618,259.078414 216.099748,244.166421 C216.099748,213.150778 241.242989,188.007632 272.258632,188.007632 L371.119632,188.007632 C379.28704,188.480415 386.935401,192.166372 392.393842,198.260263 L474.869789,300.324632 L474.978789,300.325211 L474.952789,300.426632 L474.953158,300.427737 L474.951789,300.427632 L474.173714,303.448935 C446.939817,406.220475 353.906452,478.028529 247.319105,478.080211 C118.007789,478.080211 12.8157895,373.144526 12.8157895,244.166421 C12.8157895,115.188316 117.905263,10.2526316 247.319105,10.2526316 L330.391053,10.2526316 Z"
        />
      </g>
    </g>
  </svg>
  <h3
    className="sc-htpNat iuKvGU"
  >
    GitHub Notifications 
    <br />
     in your menu bar.
  </h3>
  <button
    aria-label="Login with GitHub"
    className="sc-bxivhb lgDNln"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="octicon"
      height={16}
      role="img"
      style={
        Object {
          "display": "inline-block",
          "fill": "currentColor",
          "userSelect": "none",
          "verticalAlign": "text-bottom",
        }
      }
      viewBox="0 0 16 16"
      width={16}
    >
      <path
        d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"
        fillRule="evenodd"
      />
    </svg>
    <span>
      Login to GitHub
    </span>
  </button>
  <button
    aria-label="Login with GitHub Enterprise"
    className="sc-bxivhb lgDNln"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="octicon"
      height={16}
      role="img"
      style={
        Object {
          "display": "inline-block",
          "fill": "currentColor",
          "userSelect": "none",
          "verticalAlign": "text-bottom",
        }
      }
      viewBox="0 0 16 16"
      width={16}
    >
      <path
        d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"
        fillRule="evenodd"
      />
    </svg>
    <span>
      Login to GitHub Enterprise
    </span>
  </button>
</div>
`;
